commit 4f1be3a8b3889de4d4117d30902f2ae385c786e7
Author: Thiago Marcos P. Santos <thiago@mapbox.com>
Date:   Thu May 5 03:21:27 2016 +0300

    [core] Fix build on gcc4
    
    - Apparently that is a gcc4 bug causing the move constructor
      to be called instead of doing a list initialization.
    
    - Ignore some compilation warnings.

diff --git a/include/mbgl/util/feature.hpp b/include/mbgl/util/feature.hpp
index c9dbc31..5e0a8f2 100644
--- a/include/mbgl/util/feature.hpp
+++ b/include/mbgl/util/feature.hpp
@@ -10,8 +10,15 @@ using Value = mapbox::geometry::value;
 
 class Feature : public mapbox::geometry::feature<double> {
 public:
+
+#if !defined(__GNUC__) || __GNUC__ >= 5
     Feature(geometry_type&& geometry_)
         : mapbox::geometry::feature<double> { std::move(geometry_) } {}
+#else
+    Feature(geometry_type&& geometry_) {
+        geometry = std::move(geometry_);
+    }
+#endif
 
     optional<Value> id {};
 };
diff --git a/src/mbgl/text/glyph_pbf.cpp b/src/mbgl/text/glyph_pbf.cpp
index eab9725..fcabc07 100644
--- a/src/mbgl/text/glyph_pbf.cpp
+++ b/src/mbgl/text/glyph_pbf.cpp
@@ -10,7 +10,10 @@
 #include <mbgl/util/token.hpp>
 #include <mbgl/util/url.hpp>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshadow"
 #include <protozero/pbf_reader.hpp>
+#pragma GCC diagnostic pop
 
 namespace {
 
diff --git a/src/mbgl/tile/vector_tile.hpp b/src/mbgl/tile/vector_tile.hpp
index 153d7dc..d88f538 100644
--- a/src/mbgl/tile/vector_tile.hpp
+++ b/src/mbgl/tile/vector_tile.hpp
@@ -3,7 +3,11 @@
 
 #include <mbgl/tile/geometry_tile.hpp>
 #include <mbgl/map/tile_id.hpp>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshadow"
 #include <protozero/pbf_reader.hpp>
+#pragma GCC diagnostic pop
 
 #include <map>
 #include <unordered_map>
